frontend/src/components/common/Resource/Resource.tsx

```diff
@@ -894,3 +895,187 @@ export function ConditionsSection(props: { resource: KubeObjectInterface | null
     </SectionBox>
   );
 }
+
+export interface VolumeSectionProps {
+  resource: KubeObjectInterface | null;
+}
+
+export interface VolumeRowsProps {
+  volume: KubeVolume;
+}
+
+export interface PrintVolumeLinkProps {
+  volumeName: string;
+  volumeKind: string;
+  volume: KubeVolume;
+}
+
+export interface VolumePrints {
+  directPrint?: any[];
+  yamlPrint?: any[];
+}
+
+export function VolumeSection(props: VolumeSectionProps) {
+  const { t } = useTranslation();
+  const { resource } = props;
+  const volumes = resource?.spec?.volumes;
+
+  if (!volumes) {
+    return null;
+  }
+
+  const namespace = resource?.metadata?.namespace;
+
+  /*
+   * printVolumeLink will print a working link that is set within the router using fields from the resource as params
+   */
+  function PrintVolumeLink(props: PrintVolumeLinkProps) {
+    const { volumeName, volumeKind, volume } = props;
+    const resourceClasses = ResourceClasses;
+    const classList = Object.keys(ResourceClasses);
+
+    for (const kind of classList) {
+      if (kind.toLowerCase() === volumeKind.toLowerCase()) {
+        const volumeClass = resourceClasses[kind];
+        const volumeRoute = volumeClass.detailsRoute;
+        const volumeNamespace = volumeClass.isNamespaced ? namespace : null;
+        // we use the keys of the volume[volumeKind], which is the non class created volume item, to find the name of the volume source
+        const volumeKeys = Object.keys(volume[volumeKind]);
+        let detailName;
+
+        const volumeKindNames = {
+          configMap: 'name',
+          secret: 'secretName',
+          persistentVolumeClaim: 'claimName',
+        };
+
+        // where the volume[volumeKind] contains a "name" word in the keys, grab that value
+        for (const key of volumeKeys) {
+          // use the volumeKindNames object to get the correct key name for the volumeKind
+          if (volumeKindNames[volumeKind as keyof typeof volumeKindNames] === key) {
+            detailName = volume[volumeKind][key];
+            // using if render fixes empty source field issue for items like emptyDir where name is = ''
+            if (detailName) {
+              return (
+                <Link
+                  routeName={volumeRoute}
+                  params={{ namespace: volumeNamespace, name: detailName }}
+                >
+                  {volumeName}
+                </Link>
+              );
+            }
+          }
+        }
```

Comment:
Unless I misunderstood it, I think the logic would be simpler if you check directly whether the volumeKindName mapped key exists in the volume definition. Something like:
```typescript
let volumeNameKey = volumeKindNames[volumeKind as keyof typeof volumeKindNames];
if (!!volumeNameKey) {
  let detailName = volume[volumeKind][volumeNameKey];
  if (!!detailName) {
    return (
      <Link .... 
    );
  }
}
```